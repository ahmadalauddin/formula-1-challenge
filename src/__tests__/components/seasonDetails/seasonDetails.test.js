import React from "react";
import {shallow} from "enzyme";
import SeasonDetails from "../../../components/seasonDetails/seasonDetails";

//mock to handle useNavigate used in the component
const mockedUsedLocation = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => mockedUsedLocation,
}));

let wrapper;
const seasonYear = ""

beforeEach(()=>{
  wrapper = shallow(<SeasonDetails year={seasonYear}/>);
})

describe("Testing season details", () => {
  it("Checking the snapshot of the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Checking if no records message shown", () => {
    expect(wrapper.find("DataNotFound")).toBeTruthy();
  });

  it("Checking if thr right season is passed, show Loader", () => {
    expect(wrapper.find("Loader")).toBeTruthy()
  });


});
