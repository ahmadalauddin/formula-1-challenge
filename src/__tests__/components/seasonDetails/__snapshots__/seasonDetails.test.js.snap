// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing season details Checking the snapshot of the component 1`] = `<DataNotFound />`;
